src/System.Management.Automation/engine/hostifaces/History.cs

```diff
@@ -1041,114 +1019,137 @@ public string Id
         /// </summary>
         protected override void EndProcessing()
         {
-            // Invoke-history can execute only one command. If multiple
-            // ids were provided, throw exception
-            if (_multipleIdProvided)
-            {
-                ThrowTerminatingError(
-                    new ErrorRecord(
-                        new ArgumentException(HistoryStrings.InvokeHistoryMultipleCommandsError),
-                        "InvokeHistoryMultipleCommandsError",
-                        ErrorCategory.InvalidArgument,
-                        targetObject: null));
-            }
-
             var ctxRunspace = (LocalRunspace)Context.CurrentRunspace;
             History history = ctxRunspace.History;
             Dbg.Assert(history != null, "History should be non null");
 
-            // Get the history entry to invoke
-            HistoryInfo entry = GetHistoryEntryToInvoke(history);
-            string commandToInvoke = entry.CommandLine;
-
-            if (!ShouldProcess(commandToInvoke))
+            // If the user didn't provide an Id, get the ID for the most
+            // recent entry
+            if (Id == null)
             {
-                return;
-            }
+                try
+                {
+                    Id = new string[1] { history.GetEntries(0, 1, true)[0].Id.ToString() };
+                }
+                catch (System.IndexOutOfRangeException)
+                {
+                    // History is empty, throw a more descriptive error
+                    Exception ex =
+                        new InvalidOperationException
+                        (
+                            StringUtil.Format(HistoryStrings.NoLastHistoryEntryFound)
+                        );
 
-            // Check if there is a loop in invoke-history
-            if (history.PresentInInvokeHistoryEntrySet(entry))
-            {
-                ThrowTerminatingError(
-                    new ErrorRecord(
-                        new InvalidOperationException(HistoryStrings.InvokeHistoryLoopDetected),
-                        "InvokeHistoryLoopDetected",
-                        ErrorCategory.InvalidOperation,
-                        targetObject: null));
-            }
-            else
-            {
-                history.AddToInvokeHistoryEntrySet(entry);
+                    ThrowTerminatingError
+                    (
```

Comment:
I know this is the style used a lot in this file, but we prefer to the below style in this code base.
So appreciate if you can use the below style :)
```
                    ThrowTerminatingError(
                        new ErrorRecord(
                            new InvalidOperationException(HistoryStrings.InvokeHistoryLoopDetected),
                            "InvokeHistoryLoopDetected",
                            ErrorCategory.InvalidOperation,
                            targetObject: null));
```